packages/block-editor/src/components/block-draggable/index.native.js

```diff
@@ -133,56 +120,29 @@ const BlockDraggableWrapper = ( { children } ) => {
 		} );
 
 		const foundClientId = blockLayout?.clientId;
-		currentClientId.current = foundClientId;
+		currentBlockLayout.current = blockLayout;
 		if ( foundClientId ) {
 			startDraggingBlocks( [ foundClientId ] );
-
-			const isBlockOutOfScrollView = blockLayout.y < scroll.offsetY.value;
-			// If the dragging block is out of the scroll view, we have to
-			// scroll the block list to show the origin position of the block.
-			if ( isBlockOutOfScrollView ) {
-				scrollAnimation.value = scroll.offsetY.value;
-				const scrollOffsetTarget = Math.max(
-					0,
-					blockLayout.y - EXTRA_OFFSET_WHEN_CLOSE_TO_TOP_EDGE
-				);
-				scrollAnimation.value = withTiming(
-					scrollOffsetTarget,
-					{ duration: SCROLL_ANIMATION_DURATION },
-					() => startScrolling( position.y )
-				);
-			} else {
-				runOnUI( startScrolling )( position.y );
-			}
+			runOnUI( startScrolling )( position.y );
 		} else {
 			// We stop dragging if no block is found.
 			runOnUI( stopDragging )();
 		}
 	};
 
 	const onStopDragging = () => {
-		if ( currentClientId.current ) {
+		if ( currentBlockLayout.current ) {
 			onBlockDrop( {
 				// Dropping is only allowed at root level
 				srcRootClientId: '',
-				srcClientIds: [ currentClientId.current ],
+				srcClientIds: [ currentBlockLayout.current.clientId ],
```

Comment:
My assumption here was that every block layout will have a `clientId`, but now that you mention it, probably is not a bad idea to add an extra check just in case 😅.

Reply:
> My assumption here was that every block layout will have a `clientId`.

Yeah for sure. 

It might be me overthinking it 😅 We could merge it like this and test it out and see if we find any issues.

Reply:
I finally applied the suggestion in https://github.com/WordPress/gutenberg/pull/40104/commits/a138dff8f7d30151344f6d8afe949e91b73ee74a.